internal/aws/cwlogs/cwlog_client_test.go

```diff
@@ -517,53 +517,74 @@ func TestUserAgent(t *testing.T) {
 	logger := zap.NewNop()
 
 	tests := []struct {
-		name                 string
-		buildInfo            component.BuildInfo
-		logGroupName         string
-		expectedUserAgentStr string
+		name                      string
+		buildInfo                 component.BuildInfo
+		logGroupName              string
+		enhancedContainerInsights bool
+		expectedUserAgentStr      string
 	}{
 		{
 			"emptyLogGroup",
 			component.BuildInfo{Command: "opentelemetry-collector-contrib", Version: "1.0"},
 			"",
+			false,
 			"opentelemetry-collector-contrib/1.0",
 		},
 		{
 			"buildInfoCommandUsed",
 			component.BuildInfo{Command: "test-collector-contrib", Version: "1.0"},
 			"",
+			false,
 			"test-collector-contrib/1.0",
 		},
 		{
 			"non container insights",
 			component.BuildInfo{Command: "opentelemetry-collector-contrib", Version: "1.1"},
 			"test-group",
+			false,
 			"opentelemetry-collector-contrib/1.1",
 		},
 		{
 			"container insights EKS",
 			component.BuildInfo{Command: "opentelemetry-collector-contrib", Version: "1.0"},
 			"/aws/containerinsights/eks-cluster-name/performance",
+			false,
 			"opentelemetry-collector-contrib/1.0 (ContainerInsights)",
 		},
 		{
 			"container insights ECS",
 			component.BuildInfo{Command: "opentelemetry-collector-contrib", Version: "1.0"},
 			"/aws/ecs/containerinsights/ecs-cluster-name/performance",
+			false,
 			"opentelemetry-collector-contrib/1.0 (ContainerInsights)",
 		},
 		{
 			"container insights prometheus",
 			component.BuildInfo{Command: "opentelemetry-collector-contrib", Version: "1.0"},
 			"/aws/containerinsights/cluster-name/prometheus",
+			false,
+			"opentelemetry-collector-contrib/1.0 (ContainerInsights)",
+		},
+		{
+			"enhanced container insights EKS",
+			component.BuildInfo{Command: "opentelemetry-collector-contrib", Version: "1.0"},
+			"/aws/containerinsights/eks-cluster-name/performance",
+			true,
+			"opentelemetry-collector-contrib/1.0 (EnhancedContainerInsights)",
+		},
+		{
+			"negative - enhanced container insights ECS",
+			component.BuildInfo{Command: "opentelemetry-collector-contrib", Version: "1.0"},
+			"/aws/containerinsights/eks-cluster-name/performance",
```

Comment:
Is this meant to be a ECS test case? I see it says `eks-cluster-name`. Also, ECS has a slightly diff log group pattern typically.
Also, this case and the prior one have the same log group name and both have enhancedContainerInsights set to `true` - so confused how the UserAgent is diff.

Reply:
yeah, this is a bad test.  Oops